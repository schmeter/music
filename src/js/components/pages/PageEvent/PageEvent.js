import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';

import Page from '../../Page';
import Image from '../../Image';

const PageEvent = ({ events }) => {
    const eventsSorted = events.sort((a, b) => {
        if (a.date < b.date) {
            return 1;
        } else if (a.date > b.date) {
            return -1;
        } else {
            return 0;
        }
    });

    return (
        <Page id="event">
            {eventsSorted.map(event => !event.hidden && (
                <div
                    key={`${event.date}${event.title}`}
                    className="event"
                >
                    <h2 className="date">
                        {format(new Date(event.date), 'dd.MM.yyyy')}
                        &nbsp;
                        {event.title}
                    </h2>
                    <h3>{event.artist.title}</h3>
                    {event.imgPath && (
                        <p>
                            <Image
                                src={event.imgPath}
                                alt={event.title}
                            />
                        </p>
                    )}
                </div>
            ))}
        </Page>
    );
};

PageEvent.propTypes = {
    events: PropTypes.array.isRequired,
};

export default PageEvent;
