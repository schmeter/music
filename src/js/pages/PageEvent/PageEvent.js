import React from 'react';
import PropTypes from 'prop-types';

import Event from './Event';
import Page from '../Page';
import { getEventId } from '../../services/event';

const PageEvent = ({
  events,
}) => {
  const eventsSorted = events.sort((a, b) => {
    if (a.date < b.date) {
      return 1;
    } else if (a.date > b.date) {
      return -1;
    } else {
      return 0;
    }
  });

  return (
    <Page id="event">
      {eventsSorted.map(event => {
        return !event.hidden && (
          <Event
            key={getEventId(event)}
            event={event}
          />
        );
      })}
    </Page>
  );
};

PageEvent.propTypes = {
  events: PropTypes.array.isRequired,
};

export default PageEvent;
