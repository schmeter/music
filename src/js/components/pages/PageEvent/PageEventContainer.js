import { connect } from 'react-redux';

import PageEvent from './PageEvent';
import { getEvents } from '../../../state/selectors/event';

const mapStateToProps = state => ({
    events: getEvents(state),
});

export default connect(mapStateToProps, null)(PageEvent);
