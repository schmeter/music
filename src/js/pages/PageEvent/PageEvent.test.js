import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PageEvent from './PageEvent';

// eslint-disable-next-line react/display-name
jest.mock('../../components/Page', () => () => <div />);

describe('PageEvent', () => {
    it('renders correctly', () => {
        const props = {
            events: [{
                title: 'test',
                date: '2020-02-02',
                artist: {
                    title: 'test',
                },
                imgPath: 'test',
            }, {
                title: 'test',
                date: '2020-02-01',
                artist: {
                    title: 'test',
                },
            }, {
                hidden: true,
                title: 'test',
                date: '2020-02-03',
                artist: {
                    title: 'test',
                },
            }, {
                title: 'test',
                date: '2020-02-03',
                artist: {
                    title: 'test',
                },
            }],
        };
        const component = shallow(<PageEvent {...props} />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
