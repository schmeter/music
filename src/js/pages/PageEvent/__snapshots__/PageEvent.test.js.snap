// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageEvent renders correctly 1`] = `
<Component
  id="event"
>
  <Event
    event={
      Object {
        "artist": Object {
          "title": "test",
        },
        "date": "2020-02-03",
        "title": "test",
      }
    }
    key="test 2020-02-03 test"
  />
  <Event
    event={
      Object {
        "artist": Object {
          "title": "test",
        },
        "date": "2020-02-02",
        "imgPath": "test",
        "title": "test",
      }
    }
    key="test 2020-02-02 test"
  />
  <Event
    event={
      Object {
        "artist": Object {
          "title": "test",
        },
        "date": "2020-02-01",
        "title": "test",
      }
    }
    key="test 2020-02-01 test"
  />
</Component>
`;
